### PR TITLE
New Secdo integration

### PR DESCRIPTION
**Pull request by dem-evgeny**

### CODE REVIEW

File: Integrations/integration-secdo.yml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  required: false
+- display: Company name
+  name: company
+  defaultvalue: ""
+  type: 0
+  required: true
+script:
+  script: |+
+    var sendRequest = function() {
+        var requestUrl = 'https://' + params.server + '/publicapi/v2/load_ioc/';
```
Comment (dorsha) [inline Integrations/integration-secdo.yml]: you should add `https://` only if not exists in the server param

File: Integrations/integration-secdo.yml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+description: Secdo's automated incident response platform hunts threats in real time
+  and delivers an endpoint detection and response solution.
+configuration:
+- display: Secdo Server
+  name: server
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: Secdo Key
```
Comment (dorsha) [inline Integrations/integration-secdo.yml]: Change to Secdo API Key

File: Integrations/integration-secdo.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        } else {
+            md += 'Opps, it seems that there is some error with adding the IOCs to Secdo.\n'
+        }
+
+        return ( {'ContentsFormat': formats['json'], 'Type': entryTypes['note'], 'Contents': {}, "HumanReadable": md} );
+    };
+
+    switch (command) {
+        case 'test-module':
+            return 'ok';
```
Comment (dorsha) [inline Integrations/integration-secdo.yml]: you should add actual check...

File: Integrations/integration-secdo.yml [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    };
+
+
+    var addIocs = function() {
+        var md = "## Secdo\n";
+        var res =  sendRequest();
+        if (res.StatusCode === 200) {
+            md += 'The IOCs have been successfully added to Secdo.\n';
+        } else {
+            md += 'Opps, it seems that there is some error with adding the IOCs to Secdo.\n'
```
Comment (dorsha) [inline Integrations/integration-secdo.yml]: I think you handle this already in the sendRequest.
Also, it is better to throw error

File: Integrations/integration-secdo.yml [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        }
+
+        return ( {'ContentsFormat': formats['json'], 'Type': entryTypes['note'], 'Contents': {}, "HumanReadable": md} );
```
Comment (dorsha) [inline Integrations/integration-secdo.yml]: no need for the surrounded ()

Comment (dorsha) [inline Integrations/integration-secdo.yml]: Change to:
```
return { ContentsFormat: formats.json, Type: entryTypes.note, Contents: {}, HumanReadable: md };
```

File: Integrations/integration-secdo.yml [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+      predefined:
+      - mixed
+      - md5hash
+      - filename
+      - fullpath
+      - dest_ip
+      - hostname
+      description: Type of artifacts
+      defaultValue: mixed
+    - name: responce
```
Comment (dorsha) [inline Integrations/integration-secdo.yml]: should be: response

Comment (dorsha) [reply Integrations/integration-secdo.yml]: and it the above code you are using it as `response`, which means that you never tested this code

File: Integrations/integration-secdo.yml [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+      required: true
+      auto: PREDEFINED
+      predefined:
+      - REPORT
+      - SYSLOG
+      - CASE
+      - WHITELIST
+      - SILENT_REPORT
+      description: How Secdo will responde after the IOCs add
```
Comment (dorsha) [inline Integrations/integration-secdo.yml]: I think the end of the sentence is wrong

File: Integrations/integration-secdo.yml [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+      auto: PREDEFINED
+      predefined:
+      - LOW
+      - MEDIUM
+      - HIGH
+      description: Severity
+      defaultValue: LOW
+    - name: comment
+      required: true
+      description: Comment
```
Comment (dorsha) [inline Integrations/integration-secdo.yml]: Should have better description

File: Integrations/integration-secdo.yml [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+    - name: data
+      required: true
+      description: The IOCs data in text (Each IOCs in seperated with new line - \n)
```
Comment (dorsha) [inline Integrations/integration-secdo.yml]: you have typos here and I would change the sentence.
seperated should be separated
and this sentence is wrong: "Each IOCs in seperated with new line"

File: Integrations/integration-secdo.yml [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+    description: Add IOCs to Secdo
```
Comment (dorsha) [inline Integrations/integration-secdo.yml]: should have better description

File: Integrations/integration-secdo.yml [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+
+        return res;
+    };
+
+
+    var addIocs = function() {
+        var md = "## Secdo\n";
+        var res =  sendRequest();
+        if (res.StatusCode === 200) {
+            md += 'The IOCs have been successfully added to Secdo.\n';
```
Comment (dorsha) [inline Integrations/integration-secdo.yml]: Not sure about that sentence... should not start with: "The IOCs.."

Comment (dorsha) [reply Integrations/integration-secdo.yml]: I also think the result should includes more information

**Comment by dem-evgeny** [comment]
About "Test playbook"
@dorsha Did you saw my Slack msg? We don't have the product. It was developed on the customer's env.

**Comment by dorsha** [comment]
I saw, you can still add tests with mock server.

File: Integrations/integration-secdo.yml (hunk 2 above)
Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: Changed

File: Integrations/integration-secdo.yml (hunk 1 above)
Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: Added

File: Integrations/integration-secdo.yml (hunk 11 above)
Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: @dorsha To what change the sentence? The Secdo doesn't respond with any information. Just ok if it submitted successfully.

File: Integrations/integration-secdo.yml (hunk 4 above)
Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: You right. Changed.

File: Integrations/integration-secdo.yml (hunk 5 above)
Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: Changed

Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: Changed

File: Integrations/integration-secdo.yml (hunk 11 above)
Comment (dorsha) [reply Integrations/integration-secdo.yml]: So just remove the "The"

File: Integrations/integration-secdo.yml (hunk 3 above)
Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: The API doesn't have checks. Only submitting new IOCs.

File: Integrations/integration-secdo.yml (hunk 6 above)
Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: Changed. And yes I do. At the customer. The customer itself tested it.

File: Integrations/integration-secdo.yml (hunk 7 above)
Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: @dorsha Why? How do you want to change it?

File: Integrations/integration-secdo.yml (hunk 8 above)
Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: Changed

File: Integrations/integration-secdo.yml (hunk 3 above)
Comment (dorsha) [reply Integrations/integration-secdo.yml]: Mmmm maybe there is another option to test connectivity..
Maybe add wrong ioc and check that the error you get is the one you expecting for (and not connection error)

File: Integrations/integration-secdo.yml (hunk 9 above)
Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: Changed

File: Integrations/integration-secdo.yml (hunk 10 above)
Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: @dorsha What do you think of?

File: Integrations/integration-secdo.yml (hunk 6 above)
Comment (dorsha) [reply Integrations/integration-secdo.yml]: So how did it work with this typo? It can't.

File: Integrations/integration-secdo.yml (hunk 3 above)
Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: There is no wrong IOC. It just pushes every text that you write to it.

File: Integrations/integration-secdo.yml (hunk 7 above)
Comment (dorsha) [reply Integrations/integration-secdo.yml]: First you start the sentence with a question word and there is no question mark.
Second "IOCs add" sounds wrong. Check with Maya

File: Integrations/integration-secdo.yml (hunk 6 above)
Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: Yeah Starnge... But it worked :)

File: Integrations/integration-secdo.yml (hunk 3 above)
Comment (dorsha) [reply Integrations/integration-secdo.yml]: Let's talk about this one

File: Integrations/integration-secdo.yml (hunk 7 above)
Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: Changed

File: Integrations/integration-secdo.yml (hunk 3 above)
Comment (dem-evgeny) [reply Integrations/integration-secdo.yml]: Let's talk. Where are you?

**Comment by dorsha** [comment]
@dem-evgeny what is the status of this PR?
I didn't see you push anything after my comments. And the doc is still missing.

**Comment by dem-evgeny** [comment]
@dorsha Yes, I fixed them on my env. Waiting for some response from the customer and after it will push it

**Comment by dorsha** [comment]
👍

**Comment by dem-evgeny** [comment]
**Overview:**
Secdo's automated incident response platform hunts threats in real time and delivers an endpoint detection and response solution.

**To set up [integration name] to work with Demisto:**
Will add after customer update

**To set up the integration on Demisto:**
1. Go to Settings => Integrations => Servers & Services => Search for Secdo
2. Click "Add instance"
3. Fill the server name (without http, just the server name or ip)
4. Fill the Secdo API key that you created in the Secdo application.
5. Fill the Company name that you created in the Secdo application.
6. Mark Do not validate server certificate (insecute) if needed.
7. Mark Use system proxy settings if needed.
8. Click "Done"

**Top Use-cases:**
Adding new IOC
1. Upload text file (with IOCs) to the war room
2. Read the file contents to the Context: !ReadFile entryID=31@25
3. Add the text to the Secdo: !secdo-add-IOCs artifact=mixed comment="My Test" data=${FileData} response=CASE severity=MEDIUM   

**Commands:**
secdo-add-IOCs - Add's IOCs to Secdo

Input:
artifact - Type of artifacts (mixed,md5hash,filename,fullpath,dest_ip,hostname)
response - The Secdo response after the IOCs were added (REPORT,SYSLOG,CASE,WHITELIST,SILENT_REPORT)
severity - Severity (LOW,MEDIUM,HIGH)
comment - Comment to the added IOCs.
data - The IOCs data in text (Each IOCs in separated a with new line - \n)

**Comment by dorsha** [comment]
@mayairv please make sure to take the documentation from this issue. So it won't get lost.

**Comment by mayairv** [comment]
@dem-evgeny Please review  this draft as I don't see this integration in the GUI
https://support.demisto.com/hc/en-us/articles/115004211234

**Comment by dem-evgeny** [comment]
@mayairv 
From there is this:
"Locate ‘ICEBRG’ by searching for it using the search box on the top of the page."? This is wrong
"(Each IOCs in separated a with new line - \n)." - I think should be  "(Each IOCs in separated with a new line - \n)."

Except this, everything looks good.

**Comment by mayairv** [comment]
fixed

**Comment by dem-evgeny** [comment]
@mayairv Thanks :)

**Pull request merged**